packages/manager/modules/telecom-dashboard/src/ftth-eligibility/telecom-dashboard-ftth-eligibility.controller.js

```diff
@@ -0,0 +1,130 @@
+import {
+  ELIGIBILITY,
+  ACCESS_TYPE,
+  CLOSURE_DETAIL_URL,
+} from './telecom-dashboard-ftth-eligibility.constant';
+
+export default class FtthEligibilityCtrl {
+  /* @ngInject */
+  constructor(coreURLBuilder, FtthEligibilityService, TucToastError) {
+    this.coreURLBuilder = coreURLBuilder;
+    this.FtthEligibilityService = FtthEligibilityService;
+    this.TucToastError = TucToastError;
+    this.ELIGIBILITY = ELIGIBILITY;
+    this.ACCESS_TYPE = ACCESS_TYPE;
+  }
+
+  $onInit() {
+    this.links = {
+      accessList: this.coreURLBuilder.buildURL('telecom', '#/xdsl-access-list'),
+      copperClosureMore: CLOSURE_DETAIL_URL,
+    };
+    this.amountServicesDisplayed = 5;
+    this.servicesLength = 0;
+
+    this.getAllServices();
+  }
+
+  getAllServices() {
+    this.services = [];
+    this.FtthEligibilityService.getAllServices()
+      .then(({ data }) => {
+        this.servicesLength = data.length;
+        if (data.length <= this.amountServicesDisplayed) {
+          this.createServiceList(data);
+        } else {
+          // Get only this.amountServicesDisplayed elements from the list
+          const limitedServices = data.slice(0, this.amountServicesDisplayed);
+          this.createServiceList(limitedServices);
+        }
+      })
+      .catch(({ data }) => {
+        this.TucToastError(data.message);
+      });
+  }
+
+  createServiceList(services) {
+    // Retrieve service info for each service
+    services.forEach((serviceName) => {
+      this.FtthEligibilityService.getService(serviceName)
+        .then(({ data }) => {
+          if (data.accessType !== 'ftth') {
+            return this.getListFiberEligibilities(serviceName, data);
+          }
+          const { accessName, accessType, description, packName } = data;
+          const createService = {
+            accessName,
+            accessType,
+            description,
+            packName,
+            migrationAvailable: this.ELIGIBILITY.not_concerned,
+            accessLink: this.coreURLBuilder.buildURL(
+              'telecom',
+              `#/pack/${packName}/xdsl/${accessName}`,
+            ),
+            migrationLink: this.coreURLBuilder.buildURL(
+              'telecom',
+              `#/pack/${packName}/migration`,
+            ),
+          };
+          return createService;
```

Comment:
This block of code seem duplicate several time. I think we can create one reusable function.